pkg/mcastmanager/mcastmanager.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mcastmanager
+
+import (
+	"github.com/cilium/cilium/pkg/addressing"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/multicast"
+)
+
+var (
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "mcast-manager")
+)
+
+// TODO(ygui): Garbage collection
```

Comment:
What does that include exactly? Do you plan to follow-up on this?

Reply:
Ok, please follow-up or open an issue so we can track and implement this eventually.

Reply:
#12538 is created.

Reply:
What are the consequences for poor garbage collection here? Would it potentially interfere with reuse of an IP address on another node, potentially causing traffic to the reused address to arrive at the local node instead of the node where the address was reused?

Reply:
The impact is minimal. The node will receive ND message that it is not supposed to receive, and it will be ignored. The traffic won't come to the node.